lib/pf/action.pm

```diff
@@ -166,14 +166,24 @@ sub action_delete_all {
 
 # TODO what is that? Isn't it dangerous?
 sub action_api {
-    my ($mac, $vid, $external_id) = @_;
+    my ($mac, $vid) = @_;
+    my $logger = Log::Log4perl::get_logger('pf::action');
     my $class_info = class_view($vid);
-    my @args =
-      (
-       $Config{'paths'}{ 'external' . $external_id },
-       $mac, $class_info->{'description'}
-    );
-    system(@args);
+    my @params = split(' ', $class_info->{'external_command'});
+    my $return;
+    my $node_info = node_view($mac);
+    my $ip = pf::iplog::mac2ip($mac) || 0;
+    $node_info = {%$node_info, 'last_ip' => $ip};
+    foreach my $param (@params) {
```

Comment:
I think this loop deserves some commenting as to what it does and how it does it so we know what we can use as params and what it does to look them up